### DISCUSSION TITLE
Decoding html Text

### DISCUSSION THREAD

**Discussion by mahdyshabeeb** [pymupdf/PyMuPDF#2713] [Looking for help] [answered]
Dear fitz community,

I have a question regarding interpreting the output of the `page.get_text()` method when using the "html" option.

I'm trying to get all the text and the images of a certain page and that's why I'm using this method. However, I am trying to pass the resulting images to the Image.open() method of the python PIL library but this is not working. I am applying the following steps:

1. Using `page.get_text("html")` to get the html representation of the page
2. Parsing the html output with BeatifulSoup
3. Getting all Tags of the html content whose name is "img" (i.e. the images)
4. Now for each one of these images, I am converting it to bytes
5. Finally I pass the resulting bytes to the `PIL.Image.open()` method.

Is there any special preprocessing needed or a correct way to convert these html image representations to bytes?

**Comment by JorjMcKie** [comment, marked as answer]
Images are embedded in the generated html source like this:
![grafik](https://github.com/pymupdf/PyMuPDF/assets/8290722/db235052-c89d-4b84-8d9c-d8df0fa04e93)
This means that the string (!) following "base64," is encoded in base64 format. You must parse that string completely and decode it (e.g. binascii module or b64_decode) into your `bytes` object.
Then define `fp = io.BytesIO(obj.getvalue())` and do `img = PIL.Image.open(fp)`.

**Reply by mahdyshabeeb** [reply]
Great! This works. Thank you! :)
